src/components/Stories/Stories.tsx

```diff
@@ -69,94 +51,50 @@ export function Stories({
     );
 
     const handleGotoPrevious = React.useCallback(() => {
-        if (currentStoryIndex > 0) {
+        setStoryIndex((currentStoryIndex) => {
+            if (currentStoryIndex <= 0) {
+                return 0;
+            }
+
             const newIndex = currentStoryIndex - 1;
-            setCurrentStoryIndex(newIndex);
             onPreviousClick?.(newIndex);
-        }
-    }, [currentStoryIndex, onPreviousClick]);
+            return newIndex;
+        });
+    }, [onPreviousClick]);
 
     const handleGotoNext = React.useCallback(() => {
-        if (currentStoryIndex < items.length - 1) {
+        setStoryIndex((currentStoryIndex) => {
+            if (currentStoryIndex >= items.length - 1) {
+                return items.length - 1;
+            }
+
             const newIndex = currentStoryIndex + 1;
-            setCurrentStoryIndex(newIndex);
             onNextClick?.(newIndex);
-        }
-    }, [currentStoryIndex, items, onNextClick]);
+            return newIndex;
+        });
+    }, [items, onNextClick]);
+
+    const indexType =
+        (items[storyIndex] === undefined && IndexType.Invalid) ||
```

Comment:
line 35 is for once checking

We can't guarantee that the index is always valid because we don't own the original array.
We can recalculate `storyIndex` if `items` will be memorized.

Reply:
and also `Invalid` - it is literally one of the statuses for this case. 
`Invalid` - `(-infinity, start) and (end, +infinity)`
`Start` - `start`
`End` - `end`
`InProgress` - `(start, end)`

Reply:
@DarkGenius what do you think about `Invalid` status? I would prefer to check it in line 35 in state initialization and in `useEffect` if `items` changes. Then `console.warn` that provided index is not in valid range and call `setCurrentStoryIndex` with [0, 0].

Everything else LGTM

Reply:
We can not do `useEffect` for `items` until `items` is not memorised. 
And if users pass this property like this `items={[...]}` it will destroy front end cause of infinite recursion.

Reply:
> @DarkGenius what do you think about `Invalid` status? I would prefer to check it in line 35 in state initialization and in `useEffect` if `items` changes. Then `console.warn` that provided index is not in valid range and call `setCurrentStoryIndex` with [0, 0].
> 
> Everything else LGTM

I think status `Invalid` is overcomplicated. Let's reset the index to a reasonable default value if the value passed is invalid

Reply:
@GermanVor yes, for better usage it should be memoized.
What do you mean by `And if users pass this property like this items={[...]} it will destroy front end cause of infinite recursion.`? If reference of items changes it will trigger `useEffect`, however if everything fine nothing would happen.

Another scenario how we could remove `Invalid` status is making `indexType` property optional, if it does not exists, then indexes are wrong and we won't render any buttons or render only close button.